whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapRelationService.java

```diff
@@ -211,35 +211,47 @@ private IpEntry searchUpResource(final IpTree ipTree, final IpInterval searchIp)
     }
 
     private IpEntry searchTopLevelResource(final IpTree ipTree, final IpInterval searchIp){
-        final List<IpEntry> lessAndExact = ipTree.findExactAndAllLessSpecific(searchIp);;
+        final List<IpEntry> lessAndExact = ipTree.findExactAndAllLessSpecific(searchIp); //Exact only if exists
 
-        for (int countLessSpecific = 0; countLessSpecific < lessAndExact.size() - 1; countLessSpecific++){
```

Comment:
lessAndExact.size() - 1;
Only works when Ip search term exist...in case the search resource doesn't exist...this just contains less
so you need to loop over all of them

Reply:
OK